### PR TITLE
Speciality Page Story with toggle

### PR DESCRIPTION
**Pull request by omonk**
## Description - [Trello ticket](https://trello.com/c/5Rd8fBwJ/542-speciality-update-create-story-for-speciality-component)

- Adds story for the entire speciality page
- Adds toggle functionality so we can hide and show specific sections of the template

### CODE REVIEW

File: src/components/Speciality/Talks.js [hunk 1]
```diff
@@ -14,7 +14,7 @@ const TalksSection = ({ talks: allTalks }) => {
   const featured = allTalks.find(({ featured }) => featured)
   const cta = allTalks.find(({ cta }) => cta)
 
-  return talks && talks.length ? (
+  return (
```
Comment (GBouffard) [inline src/components/Speciality/Talks.js]: Why did you remove `talks && talks.length`? Could have been used as `talks && talks.length && <BlueBackground>` with `talks = []` in props

File: src/templates/speciality-component.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+  const tutorials = getExternalType(flattenedSpeciality, `Tutorial`)
+  const books = getExternalType(flattenedSpeciality, `Book`)
+  const specialityEvents =
+    events && events.length > 0 ? getSpecialityEvents(title, events) : []
+
+  const renderProjectsSection =
+    validateProjects(relatedProjects) || validateClients(clients)
+  const renderCommunitySection = validateCommunity(flattenedSpeciality)
+
+  const renderTalksSection = talks && talks.length > 0
```
Comment (GBouffard) [inline src/templates/speciality-component.js]: I would have called all these sections `hasTalks`, `hasBooks` and `hasBlogs` even if it's pretty easy to understand what it does.

File: src/components/Speciality/Books.js [hunk 3]
```diff
@@ -48,66 +48,65 @@ const BooksBox = styled(ExternalAnchor)`
...
-                  </Padding>
-                </BooksBox>
-              </Padding>
-            </BookCol>
-          ))}
-        </Row>
-        <Row>
-          <Col width={[1]}>
+const BooksSection = ({ books, title }) => (
+  <Grid>
```
Comment (GBouffard) [inline src/components/Speciality/Books.js]: Same comment as below. Always a good safety net to have `books && books.length &&` with `books = []` in props

File: src/components/Speciality/Talks.js (hunk 1 above)
Comment (omonk) [reply src/components/Speciality/Talks.js]: Technically due to the updates in the `speciality-component` we don't need these but you're right they should still be localised here if we're not setting any defaults as well
